File: Habitica/src/main/java/com/habitrpg/android/habitica/models/inventory/CustomizationSet.java
package com.habitrpg.android.habitica.models.inventory;

import com.habitrpg.shared.habitica.models.inventory.Customization;

import java.util.List;

/**
 * Created by viirus on 19/01/16.
 */
public class CustomizationSet {

    public String text;
    public String identifier;
    public Integer price;
    public boolean hasPurchasable;

    public List<Customization> customizations;
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/models/inventory/CustomizationSet.kt
package com.habitrpg.android.habitica.models.inventory

class CustomizationSet {
    var text: String? = null
    var identifier: String? = null
    var price: Int = 0
    var hasPurchasable = false
    var customizations: MutableList<Customization> = mutableListOf()
    var ownedCustomizations: MutableList<Customization> = mutableListOf()

    fun isSetDeal(): Boolean {
        var total = 0
        for (customization in customizations) {
            if (!ownedCustomizations.contains(customization)) {
                customization.price?.let { total += it }
            }
        }
        return total >= price
    }
}
